fastlane_core/lib/fastlane_core/tool_collector.rb

```diff
@@ -0,0 +1,72 @@
+module FastlaneCore
+  class ToolCollector
+    TOOLS = %w(fastlane fastlane_core deliver snapshot frameit pem sigh produce cert gym pilot credentials_manager spaceship scan supply watchbuild match screengrab)
```

Comment:
Yes, I think you're right that it's better if `fastlane_core` doesn't develop knowledge about the set of tools.

I think you're also right that the way the collector is used at the tool level (rather than the action level) doesn't require the `is_official?` check.